yal/pleasecallme.py

```diff
@@ -1088,3 +1089,37 @@ async def state_website(self):
                 "menu": "state_pre_mainmenu",
             },
         )
+
+    async def state_confirm_redirect_please_call_me(self):
+        async def _next(choice: Choice):
+            if choice.value == "help now":
+                return "state_please_call_start"
+            if choice.value == "mistake":
+                if (
+                    "emergency_keyword_previous_state" in self.user.metadata
+                    and self.user.metadata["emergency_keyword_previous_state"]
+                ):
+                    return self.user.metadata["emergency_keyword_previous_state"]
+
+        user_input = self.inbound.content
+        question = self._(
+            "\n".join(
+                [
+                    f'Hi, would you like to talk to someone about "{user_input}"?',
+                    "",
+                    "----",
+                    "*Or reply:*",
+                    BACK_TO_MAIN,
+                ]
+            )
+        )
+        return WhatsAppButtonState(
+            self,
+            question=question,
+            choices=[
+                Choice("help now", "please help"),
+                Choice("mistake", "go back"),
+            ],
+            error=self._(get_generic_error()),
+            next=_next,
```

Comment:
We could maybe make this a bit simpler by using the `dict` format for next?
```
next={
  "help now": "state_please_call_start",
  "mistake": self.user.metadata["emergency_keyword_previous_state",
}
```

Reply:
@rudigiesler I was worried that if the key was missing from metadata, it might cause chaos?

Reply:
Well at the moment, if it's missing from metadata, our `next` function is returning `None`, which will still cause chaos. If we think this is possible, maybe we should add a test for it?

We can also do `self.user.,metadata.get("emergency_keyword_previous_state", "fallbackstatehere")` if you want a fallback for if that metadata isn't set